[auto] UAT stuck-loop (×9) + orphaned worktree after milestone completion

## Problem

Two cascading bugs cause severe instability in `gsd auto` across milestones:

1. **`run-uat` stuck-loop** (S01×6, S02×9, S03×4): the `uat-verdict-gate` dispatch rule blocks progression with `level: "warning"`, which triggers `pauseAuto`. When the session resumes, it re-enters the same worktree and re-dispatches `run-uat` endlessly because the `ASSESSMENT` file written by the agent doesn't survive cross-session sync.

2. **Orphaned worktree after milestone completion**: after `complete-milestone` + `mergeAndExit`, the worktree directory and `milestone/<MID>` branch are not removed. Every subsequent `/gsd auto` call discovers the orphaned worktree and re-enters it, diverging from the merged `main`.

---

## Root Cause

### Bug 1 — UAT stuck-loop: `run-uat` ASSESSMENT file not synced into worktree on resume

**Location**: `auto-worktree.ts` — `syncProjectRootToWorktree()` + `auto-post-unit.ts` — `LIFECYCLE_ONLY_UNITS`

**Sequence**:
1. `run-uat` executes inside the worktree. The agent writes `S01-ASSESSMENT.md` (verdict: `FAIL`) via `gsd_summary_save` → `saveArtifactToDb` → `process.cwd()` → disk path inside the worktree.
2. `run-uat` is in `LIFECYCLE_ONLY_UNITS` → **no auto-commit**. The file stays untracked in the worktree working tree.
3. `syncStateToProjectRoot()` (post-unit) copies the worktree `.gsd/` → project root. `S01-ASSESSMENT.md:FAIL` lands in the project root.
4. `uat-verdict-gate` sees verdict `"fail"` → `level: "warning"` → calls `pauseAuto()`.
5. On next `/gsd auto` (resume), `syncProjectRootToWorktree()` runs with **`force: false`** (additive-only). If `S01-ASSESSMENT.md` is already present in the worktree (from step 1), it is NOT overwritten — the stale `FAIL` copy blocks.
6. But **between sessions**, `syncProjectRootToWorktree` also deletes `gsd.db` in the worktree and re-creates it from disk. During that rebuild, the worktree's in-memory write from step 1 may not be on disk if a session crash preceded the pause — leaving the worktree without any `ASSESSMENT` file at all.
7. `checkNeedsRunUat` (`auto-prompts.ts:~802`) finds no verdict → re-dispatches `run-uat`.

**The compounding issue**: when the session is stopped (not merely paused) and restarts from the project root — because `paused-session.json` was not written or was lost — `enterMilestone()` in `auto-start.ts` must re-enter the worktree. It does so by calling `getAutoWorktreePath()` then `enterAutoWorktree()`. If the worktree was left in a state where `reconcilePlanCheckboxes()` can't find the `ASSESSMENT` verdict, `checkNeedsRunUat` dispatches `run-uat` again.

**Code path** (confirmed in source):
```ts
// auto-prompts.ts ~802
if (!prefs?.uat_dispatch) return null;
// ...
const assessmentFile = resolveSliceFile(base, mid, sid, "ASSESSMENT");
if (assessmentFile) {
  const assessmentContent = await loadFile(assessmentFile);
  if (assessmentContent && hasVerdict(assessmentContent)) return null; // ← only exits if file found AND has verdict
}
```

If `S01-ASSESSMENT.md` is absent from the worktree (deleted by DB rebuild), `loadFile` returns null, `hasVerdict` is never called → `run-uat` is dispatched again.

### Bug 2 — Orphaned worktree: `teardownAutoWorktree` silently fails, branch not deleted

**Location**: `auto-worktree.ts` — `teardownAutoWorktree()` + `worktree-manager.ts` — `removeWorktree()`

After `complete-milestone`, `stopAuto()` detects the SUMMARY and calls `resolver.mergeAndExit()`. The squash merge succeeds (confirmed: `feat(M011): ...` commit on `main`). However `teardownAutoWorktree()` attempts `removeWorktree()` which calls `nativeWorktreeRemove()` + `nativeBranchDelete()`. Both calls are wrapped in `try { ... } catch { /* may fail */ }` — **failures are silently swallowed**. When the worktree directory contains uncommitted files (e.g. untracked `S01-UAT-RESULT.md` written by `run-uat`), `git worktree remove` fails with "contains untracked files". The catch swallows it. The force-remove attempt also fails if the process CWD is still inside the worktree at the time of the call. Result: the worktree directory and branch remain on disk forever.

**Confirmed in production**:
```
git worktree list:
  /project                              06ff412 [main]     ← M011 merged ✓  
  /project/.gsd/worktrees/M011         5ee7af0 [milestone/M011]  ← orphaned ✗
```

**Code path** (confirmed in source):
```ts
// worktree-manager.ts ~364
try { nativeWorktreeRemove(basePath, resolvedWtPath, useForce); } catch { /* may fail */ }
// If the directory is still there, try harder:
if (existsSync(resolvedWtPath)) {
  try { nativeWorktreeRemove(basePath, resolvedWtPath, true); } catch { /* may fail */ }
}
// → nativeWorktreePrune called, but the .git file still exists → prune has no effect
// → nativeBranchDelete called but branch is still in git's worktree registry → fails
```

The `teardownAutoWorktree()` call in `auto-worktree.ts` has a warning for this case (logWarning + rmSync fallback) but the `rmSync` also silently fails when git's internal `.git/worktrees/<name>` lock file is present.

---

## Expected Behavior

### Fix 1 — UAT ASSESSMENT persistence

`run-uat` should be in `LIFECYCLE_ONLY_UNITS` for auto-commit purposes (correct — it writes only `.gsd/`) BUT `syncStateToProjectRoot()` must be called **unconditionally after every unit** including `run-uat`. Currently:

```ts
// auto-post-unit.ts ~344 — EXISTING
if (!opts?.skipWorktreeSync && s.originalBasePath && s.originalBasePath !== s.basePath) {
  await runSafely("postUnit", "worktree-sync", () => {
    syncStateToProjectRoot(s.basePath, s.originalBasePath!, s.currentMilestoneId);
  });
}
```

This already runs ✓. The real fix is in `syncProjectRootToWorktree()` on resume:

**Option A (preferred)**: When a `ASSESSMENT` file is already in the project root with `verdict: PASS`, overwrite the worktree copy regardless of `force`:
```ts
// auto-worktree.ts — syncProjectRootToWorktree()
// CHANGE: for ASSESSMENT files, use force:true when project root has a PASS verdict
// and worktree has a FAIL verdict.
```

**Option B (simpler)**: Add `run-uat` to the set of units whose output ASSESSMENT file is force-synced back on the next dispatch iteration (similar to how `completed-units.json` is force-synced).

**Option C**: After `pauseAuto` triggered by `uat-verdict-gate`, immediately call `syncStateToProjectRoot` to ensure the latest verdict propagates before the lock is released.

### Fix 2 — Orphaned worktree cleanup

`teardownAutoWorktree()` / `removeWorktree()` must not silently swallow failures. Concrete fix:

```ts
// worktree-manager.ts — removeWorktree()

// After both nativeWorktreeRemove attempts fail:
if (existsSync(resolvedWtPath)) {
  // Force-remove the .git/worktrees/<name> internal directory to release the git lock,
  // then retry the filesystem remove.
  try {
    const wtInternalDir = join(basePath, ".git", "worktrees", name);
    if (existsSync(wtInternalDir)) rmSync(wtInternalDir, { recursive: true, force: true });
    rmSync(resolvedWtPath, { recursive: true, force: true });
    nativeWorktreePrune(basePath);
  } catch (err) {
    logWarning("reconcile", `Worktree directory could not be removed: ${err}. Manual cleanup: rm -rf "${resolvedWtPath}"`, { worktree: name });
  }
}
```

Additionally, `stopAuto()` should call `syncStateToProjectRoot()` **before** attempting `mergeAndExit()` to ensure all ASSESSMENT/SUMMARY artifacts are on the project-root filesystem before teardown:

```ts
// auto.ts — stopAuto(), before resolver.mergeAndExit(...)
if (s.originalBasePath && s.basePath !== s.originalBasePath) {
  syncStateToProjectRoot(s.basePath, s.originalBasePath, s.currentMilestoneId);
}
```

---

## Environment

- GSD version: 2.53.0
- Model: claude-sonnet-4-6
- Units affected: `run-uat/M011/S01` (×6), `run-uat/M011/S02` (×9), `run-uat/M011/S03` (×4), `complete/milestone-M011`

---

## Reproduction Context

- Active milestone with worktree isolation enabled
- `uat_dispatch: true` in preferences
- First `run-uat` produces `verdict: FAIL` in ASSESSMENT
- Session pauses/stops via `uat-verdict-gate` (warning-level stop)
- User restarts `/gsd auto`
- New session cannot find verdict in worktree ASSESSMENT → re-dispatches `run-uat` indefinitely

---

## Forensic Evidence

```
[ERROR] stuck-loop: run-uat/M011/S01 dispatched 7 times (worktree journal: 1 time, project-root session: 6 times)
[ERROR] stuck-loop: run-uat/M011/S02 dispatched 9 times
[ERROR] stuck-loop: run-uat/M011/S03 dispatched 4 times

Journal trace (worktree):
  11:24:46 dispatch-match run-uat M011/S01
  11:27:48 unit-end run-uat M011/S01 → artifactVerified:true
  11:28:05 dispatch-stop reason:"UAT verdict for S01 is 'fail' — blocking progression"

Activity log trace (project root, 2nd session):
  004-run-uat-M011-S01 (ASSESSMENT written with verdict:PASS via gsd_summary_save)
  005-run-uat-M011-S01 (again — ASSESSMENT not found in worktree)
  ...009-run-uat-M011-S01

Orphaned worktree after complete-milestone:
  git worktree list showed milestone/M011 still registered after merge commit:
  "feat(M011): Inspecteur Xray CSS flottant — merge milestone/M011"
  
error-trace: ls src/lib/xrayWindow.svelte.ts: No such file or directory
  → complete-milestone running in project-root context, file only exists in worktree
```

---

*Auto-generated by `/gsd forensics` — GSD 2.53.0*

--- a/src/resources/extensions/gsd/auto-worktree.ts
+++ b/src/resources/extensions/gsd/auto-worktree.ts
@@ -102,6 +102,67 @@ function isSamePath(a: string, b: string): boolean {
   }
 }
 
+// ─── ASSESSMENT Force-Sync Helper (#2821) ─────────────────────────────────
+
+/** Regex matching YAML frontmatter `verdict:` field. */
+const VERDICT_RE = /verdict:\s*[\w-]+/i;
+
+/**
+ * Walk a milestone directory and force-overwrite ASSESSMENT files in the
+ * destination when the source copy contains a `verdict:` field.
+ *
+ * This is the targeted fix for the UAT stuck-loop (#2821): the main
+ * safeCopyRecursive uses force:false to protect worktree-authoritative
+ * files (#1886), but ASSESSMENT files written by run-uat must be
+ * forward-synced when the project root has a verdict. Without this,
+ * the worktree retains a stale FAIL or missing ASSESSMENT and
+ * checkNeedsRunUat re-dispatches run-uat indefinitely.
+ *
+ * Only overwrites when the source has a verdict — never clobbers a
+ * worktree ASSESSMENT with a verdictless project-root copy.
+ */
+function forceOverwriteAssessmentsWithVerdict(
+  srcMilestoneDir: string,
+  dstMilestoneDir: string,
+): void {
+  if (!existsSync(srcMilestoneDir)) return;
+
+  // Walk slices/<SID>/ looking for *-ASSESSMENT.md files
+  const slicesDir = join(srcMilestoneDir, "slices");
+  if (!existsSync(slicesDir)) return;
+
+  try {
+    for (const sliceEntry of readdirSync(slicesDir, { withFileTypes: true })) {
+      if (!sliceEntry.isDirectory()) continue;
+      const srcSliceDir = join(slicesDir, sliceEntry.name);
+      const dstSliceDir = join(dstMilestoneDir, "slices", sliceEntry.name);
+
+      try {
+        for (const fileEntry of readdirSync(srcSliceDir, { withFileTypes: true })) {
+          if (!fileEntry.isFile()) continue;
+          if (!fileEntry.name.endsWith("-ASSESSMENT.md")) continue;
+
+          const srcFile = join(srcSliceDir, fileEntry.name);
+          try {
+            const srcContent = readFileSync(srcFile, "utf-8");
+            if (!VERDICT_RE.test(srcContent)) continue; // no verdict in source — skip
+
+            // Source has a verdict — force-copy into worktree
+            mkdirSync(dstSliceDir, { recursive: true });
+            safeCopy(srcFile, join(dstSliceDir, fileEntry.name), { force: true });
+          } catch {
+            /* non-fatal per file */
+          }
+        }
+      } catch {
+        /* non-fatal per slice */
+      }
+    }
+  } catch {
+    /* non-fatal */
+  }
+}
+
 // ─── Module State ──────────────────────────────────────────────────────────
 
 /** Original project root before chdir into auto-worktree. */
@@ -212,6 +273,19 @@ export function syncProjectRootToWorktree(
     join(prGsd, "milestones", milestoneId),
     join(wtGsd, "milestones", milestoneId),
     { force: false },
+  );
+
+  // Force-sync ASSESSMENT files that have a verdict from project root (#2821).
+  // The additive-only copy above preserves worktree-authoritative files, but
+  // ASSESSMENT files are special: after run-uat writes a verdict and post-unit
+  // syncs it to the project root, the worktree may retain a stale copy (e.g.
+  // verdict:fail while the project root has verdict:pass from a retry). On
+  // session resume the DB is rebuilt from disk, and if the stale ASSESSMENT
+  // persists, checkNeedsRunUat finds no passing verdict → re-dispatches
+  // run-uat indefinitely (stuck-loop ×9).
+  forceOverwriteAssessmentsWithVerdict(
+    join(prGsd, "milestones", milestoneId),
+    join(wtGsd, "milestones", milestoneId),
   );
 
   // Forward-sync completed-units.json from project root to worktree.

--- a/src/resources/extensions/gsd/tests/uat-stuck-loop-orphaned-worktree.test.ts
+++ b/src/resources/extensions/gsd/tests/uat-stuck-loop-orphaned-worktree.test.ts
@@ -1,0 +1,289 @@
+/**
+ * uat-stuck-loop-orphaned-worktree.test.ts — Regression tests for #2821.
+ *
+ * Reproduces two cascading bugs:
+ *
+ * Bug 1 — UAT stuck-loop: syncProjectRootToWorktree uses force:false for
+ *   milestone files. When the project root has an ASSESSMENT with a verdict
+ *   but the worktree has a stale/empty ASSESSMENT (or none at all after DB
+ *   rebuild), the verdict is NOT synced into the worktree. checkNeedsRunUat
+ *   finds no verdict → re-dispatches run-uat indefinitely.
+ *
+ * Bug 2 — Orphaned worktree: removeWorktree silently swallows failures when
+ *   git worktree remove fails (untracked files, CWD inside worktree, etc.).
+ *   The worktree directory and branch persist on disk after teardown.
+ *   teardownAutoWorktree has a fallback rmSync but it also fails when the
+ *   git internal .git/worktrees/<name> directory holds a lock.
+ */
+
+import { describe, test, beforeEach, afterEach } from "node:test";
+import assert from "node:assert/strict";
+import {
+  mkdtempSync,
+  mkdirSync,
+  writeFileSync,
+  rmSync,
+  existsSync,
+  readFileSync,
+} from "node:fs";
+import { join } from "node:path";
+import { tmpdir } from "node:os";
+import { execFileSync } from "node:child_process";
+
+import { syncProjectRootToWorktree } from "../auto-worktree.ts";
+import {
+  createWorktree,
+  removeWorktree,
+  worktreePath,
+} from "../worktree-manager.ts";
+
+function git(args: string[], cwd: string): string {
+  return execFileSync("git", args, {
+    cwd,
+    stdio: ["ignore", "pipe", "pipe"],
+    encoding: "utf-8",
+  }).trim();
+}
+
+function makeBaseRepo(): string {
+  const base = mkdtempSync(join(tmpdir(), "gsd-2821-"));
+  git(["init", "-b", "main"], base);
+  git(["config", "user.name", "Test"], base);
+  git(["config", "user.email", "test@test.com"], base);
+  writeFileSync(join(base, "README.md"), "# test\n");
+  mkdirSync(join(base, ".gsd", "milestones", "M011"), { recursive: true });
+  git(["add", "."], base);
+  git(["commit", "-m", "init"], base);
+  return base;
+}
+
+// ─── Bug 1: ASSESSMENT force-sync ─────────────────────────────────────────
+
+describe("#2821 Bug 1 — ASSESSMENT file force-synced on resume", () => {
+  let mainBase: string;
+  let wtBase: string;
+
+  beforeEach(() => {
+    mainBase = mkdtempSync(join(tmpdir(), "gsd-2821-main-"));
+    wtBase = mkdtempSync(join(tmpdir(), "gsd-2821-wt-"));
+    mkdirSync(join(mainBase, ".gsd", "milestones", "M011", "slices", "S01"), {
+      recursive: true,
+    });
+    mkdirSync(join(wtBase, ".gsd", "milestones", "M011", "slices", "S01"), {
+      recursive: true,
+    });
+  });
+
+  afterEach(() => {
+    rmSync(mainBase, { recursive: true, force: true });
+    rmSync(wtBase, { recursive: true, force: true });
+  });
+
+  test("force-syncs ASSESSMENT with verdict from project root into worktree when worktree copy has no verdict", () => {
+    // Project root has ASSESSMENT with a PASS verdict (written by run-uat, synced by post-unit)
+    const prAssessment = join(
+      mainBase,
+      ".gsd",
+      "milestones",
+      "M011",
+      "slices",
+      "S01",
+      "S01-ASSESSMENT.md",
+    );
+    writeFileSync(
+      prAssessment,
+      "---\nverdict: pass\n---\n# S01 Assessment\nAll tests pass.\n",
+    );
+
+    // Worktree has a stale ASSESSMENT with FAIL verdict (from the initial run-uat execution)
+    const wtAssessment = join(
+      wtBase,
+      ".gsd",
+      "milestones",
+      "M011",
+      "slices",
+      "S01",
+      "S01-ASSESSMENT.md",
+    );
+    writeFileSync(
+      wtAssessment,
+      "---\nverdict: fail\n---\n# S01 Assessment\nSome tests fail.\n",
+    );
+
+    syncProjectRootToWorktree(mainBase, wtBase, "M011");
+
+    // The worktree ASSESSMENT must now have the project root's PASS verdict
+    const content = readFileSync(wtAssessment, "utf-8");
+    assert.ok(
+      content.includes("verdict: pass"),
+      `Expected worktree ASSESSMENT to have verdict:pass after sync, got: ${content.slice(0, 100)}`,
+    );
+  });
+
+  test("force-syncs ASSESSMENT from project root when worktree has no ASSESSMENT at all", () => {
+    // Project root has ASSESSMENT with verdict
+    const prAssessment = join(
+      mainBase,
+      ".gsd",
+      "milestones",
+      "M011",
+      "slices",
+      "S01",
+      "S01-ASSESSMENT.md",
+    );
+    writeFileSync(
+      prAssessment,
+      "---\nverdict: pass\n---\n# S01 Assessment\n",
+    );
+
+    // Worktree has NO ASSESSMENT (deleted during DB rebuild)
+    // — file simply doesn't exist
+
+    syncProjectRootToWorktree(mainBase, wtBase, "M011");
+
+    const wtAssessment = join(
+      wtBase,
+      ".gsd",
+      "milestones",
+      "M011",
+      "slices",
+      "S01",
+      "S01-ASSESSMENT.md",
+    );
+    assert.ok(
+      existsSync(wtAssessment),
+      "ASSESSMENT should be copied to worktree when missing",
+    );
+    const content = readFileSync(wtAssessment, "utf-8");
+    assert.ok(
+      content.includes("verdict: pass"),
+      `Synced ASSESSMENT should contain verdict:pass, got: ${content.slice(0, 100)}`,
+    );
+  });
+
+  test("does NOT overwrite worktree ASSESSMENT when project root has no verdict", () => {
+    // Project root has ASSESSMENT without verdict (incomplete)
+    const prAssessment = join(
+      mainBase,
+      ".gsd",
+      "milestones",
+      "M011",
+      "slices",
+      "S01",
+      "S01-ASSESSMENT.md",
+    );
+    writeFileSync(prAssessment, "# S01 Assessment\nIn progress...\n");
+
+    // Worktree has ASSESSMENT with verdict:fail
+    const wtAssessment = join(
+      wtBase,
+      ".gsd",
+      "milestones",
+      "M011",
+      "slices",
+      "S01",
+      "S01-ASSESSMENT.md",
+    );
+    writeFileSync(
+      wtAssessment,
+      "---\nverdict: fail\n---\n# S01 Assessment\nSome tests fail.\n",
+    );
+
+    syncProjectRootToWorktree(mainBase, wtBase, "M011");
+
+    // Worktree copy should NOT be overwritten by the verdictless project root copy
+    const content = readFileSync(wtAssessment, "utf-8");
+    assert.ok(
+      content.includes("verdict: fail"),
+      `Worktree ASSESSMENT should keep verdict:fail when project root has no verdict, got: ${content.slice(0, 100)}`,
+    );
+  });
+});
+
+// ─── Bug 2: Orphaned worktree cleanup ─────────────────────────────────────
+
+describe("#2821 Bug 2 — removeWorktree cleans up despite untracked files", () => {
+  let base: string;
+
+  beforeEach(() => {
+    base = makeBaseRepo();
+  });
+
+  afterEach(() => {
+    rmSync(base, { recursive: true, force: true });
+  });
+
+  test("removes worktree directory even when it contains untracked files", () => {
+    const info = createWorktree(base, "M011", {
+      branch: "milestone/M011",
+    });
+
+    // Simulate run-uat writing untracked files (S01-UAT-RESULT.md, ASSESSMENT)
+    mkdirSync(
+      join(info.path, ".gsd", "milestones", "M011", "slices", "S01"),
+      { recursive: true },
+    );
+    writeFileSync(
+      join(
+        info.path,
+        ".gsd",
+        "milestones",
+        "M011",
+        "slices",
+        "S01",
+        "S01-UAT-RESULT.md",
+      ),
+      "# UAT Result\nverdict: fail\n",
+    );
+    writeFileSync(
+      join(
+        info.path,
+        ".gsd",
+        "milestones",
+        "M011",
+        "slices",
+        "S01",
+        "S01-ASSESSMENT.md",
+      ),
+      "---\nverdict: fail\n---\n# Assessment\n",
+    );
+
+    removeWorktree(base, "M011", {
+      branch: "milestone/M011",
+      deleteBranch: true,
+      force: true,
+    });
+
+    const wtDir = worktreePath(base, "M011");
+    assert.ok(
+      !existsSync(wtDir),
+      `Worktree directory should be removed after teardown, but still exists at ${wtDir}`,
+    );
+  });
+
+  test("removes git internal worktree metadata after filesystem removal", () => {
+    createWorktree(base, "M011", {
+      branch: "milestone/M011",
+    });
+
+    removeWorktree(base, "M011", {
+      branch: "milestone/M011",
+      deleteBranch: true,
+      force: true,
+    });
+
+    // The git internal worktree directory should be cleaned up
+    const gitInternalWorktreeDir = join(base, ".git", "worktrees", "M011");
+    assert.ok(
+      !existsSync(gitInternalWorktreeDir),
+      `Git internal worktree dir should be removed: ${gitInternalWorktreeDir}`,
+    );
+
+    // The branch should be deleted
+    const branches = git(["branch"], base);
+    assert.ok(
+      !branches.includes("milestone/M011"),
+      "milestone/M011 branch should be deleted after removeWorktree",
+    );
+  });
+});

--- a/src/resources/extensions/gsd/worktree-manager.ts
+++ b/src/resources/extensions/gsd/worktree-manager.ts
@@ -365,6 +365,29 @@ export function removeWorktree(
     try { nativeWorktreeRemove(basePath, resolvedWtPath, true); } catch { /* may fail */ }
   }
 
+  // (#2821) If the worktree directory STILL exists after both native removal
+  // attempts (e.g. untracked files like ASSESSMENT/UAT-RESULT prevent git
+  // worktree remove), force-remove the git internal worktree metadata first,
+  // then remove the filesystem directory. Without this, the .git/worktrees/<name>
+  // lock prevents rmSync from cleaning up, and the orphaned worktree directory
+  // causes every subsequent `/gsd auto` to re-enter the stale worktree.
+  if (existsSync(resolvedWtPath)) {
+    try {
+      const wtInternalDir = join(basePath, ".git", "worktrees", name);
+      if (existsSync(wtInternalDir)) {
+        rmSync(wtInternalDir, { recursive: true, force: true });
+      }
+      rmSync(resolvedWtPath, { recursive: true, force: true });
+    } catch {
+      logWarning(
+        "reconcile",
+        `Worktree directory could not be removed after git internal cleanup: ${resolvedWtPath}. ` +
+          `Manual cleanup: rm -rf "${resolvedWtPath.replaceAll("\\", "/")}"`,
+        { worktree: name },
+      );
+    }
+  }
+
   // Prune stale entries so git knows the worktree is gone
   nativeWorktreePrune(basePath);
 

